qsirecon/interfaces/converters.py

```diff
@@ -211,6 +217,61 @@ def _run_interface(self, runtime):
         return runtime
 
 
+class _MergeFODGQIFibsInputSpec(BaseInterfaceInputSpec):
+    csd_fib_file = File(exists=True, mandatory=True)
+    reference_fib_file = File(exists=True, mandatory=True)
+    fibgz_map = File(exists=True)
+
+
+class _MergeFODGQIFibsOutputSpec(TraitedSpec):
+    fibgz = File(exists=True, mandatory=True)
+    fibgz_map = File(exists=True)
+
+
+class MergeFODGQIFibs(SimpleInterface):
+    input_spec = _MergeFODGQIFibsInputSpec
+    output_spec = _MergeFODGQIFibsOutputSpec
+
+    def _run_interface(self, runtime):
+
+        # fname presuffix doesn't work with .fib.gz
+        fib_name = Path(self.inputs.reference_fib_file).name.replace(".odf.", ".odf.FOD.")
+        merged_fib_file = str(Path(runtime.cwd) / fib_name)
+        merged_fib_file = (
+            merged_fib_file if not merged_fib_file.endswith(".gz") else merged_fib_file[:-3]
+        )
+
+        combine_gqi_and_csd_fib_files(
+            path_gqi_fib=self.inputs.reference_fib_file,
+            path_fod_fib=self.inputs.csd_fib_file,
+            merged_fib=merged_fib_file,
+        )
+
+        # gzip the merged file
+        merged_fibgz_file = merged_fib_file + ".gz"
+
+        p = subprocess.Popen(
+            ["gzip", "-v", merged_fib_file], stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        response = p.communicate()
+        if not p.returncode == 0:
+            raise Exception(f"Gzip exitted with code {p.returncode}: {response}")
+        if not Path(merged_fibgz_file).exists():
+            raise Exception(f"Failed to gzip {merged_fib_file}")
```

Comment:
python's gzip has been historically very slow and flaky with fib data. There is a function later in this script that uses zcat from the command line to open gzipped fib files because python's gzip was blowing up memory and running hundreds of times slower.